### PR TITLE
Add collections to browse

### PR DESCRIPTION
**Pull request by alejandromumo**
closes https://github.com/zenodo/zenodo-rdm/issues/1006

### CODE REVIEW

**Review by slint — COMMENTED** [review]
...part 2 ready.

Mostly simplifications/ergonomics, so nothing critical (besides the consistent naming)

File: invenio_rdm_records/collections/api.py [hunk 1]
```diff
@@ -155,108 +146,59 @@ def query(self):
...
-            )
-            .order_by(self.model_cls.path, self.model_cls.order)
-        )
-        ret = db.session.execute(stmt).scalars().all()
-        return [type(self)(r) for r in ret]
+        res = self.model_cls.get_children(self.model)
+        return [type(self)(r) for r in res]
 
-    def get_subcollections(self, max_depth=3):
+    def get_subcollections(self):
```
[the remark this answers is not in the document]
Comment (alejandromumo) [reply invenio_rdm_records/collections/api.py:192]: The idea is that the property `subcollections` masks the underlying behaviour, based on the depth (`get_subcollections` is a less performant query than `get_children`)

File: invenio_rdm_records/collections/results.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+                }
+            # Find the parent ID from the collection's path (last valid ID in the path)
+            path_parts = _c.split_path_to_ids()
+            if path_parts:
+                parent_id = path_parts[-1]
+                # Add the collection as a child of its parent
+                res[parent_id]["children"].append(_c.id)
+
+        # Add breadcrumbs
+        res[self._collection.id]["breadcrumbs"] = self.breadcrumbs
```
[the remark this answers is not in the document]
Comment (alejandromumo) [reply invenio_rdm_records/collections/results.py:91]: yes, breadcrumbs are built from root to leaf. The breadcrumbs always takes into consideration `path` and `order`, actually this is guaranteed by the underlying models (DB)

File: invenio_rdm_records/collections/api.py [hunk 3]
```diff
@@ -155,108 +146,59 @@ def query(self):
         for _a in self.ancestors:
             q += f"({_a.model.search_query}) AND "
         q += f"({self.model.search_query})"
+        Collection.validate_query(q)
```
[the remark this answers is not in the document]
Comment (alejandromumo) [reply invenio_rdm_records/collections/api.py]: I ended up implementing this. 

The community record search accepts a `collection_id` that is injected in the search UI as a `hidden_param`. Then the parameter is processed by the community records search view, the collection is resolved and the query computed - always using dsl so we can query records inside a collection.

**Pull request merged**